src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -127,17 +127,17 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
     case PredicateCondition::BetweenLowerExclusive: {
       Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
       const auto value2 = boost::get<T>(*variant_value2);
-      return value > max || value2 <= min;
+      return value >= max || value2 < min;
     }
     case PredicateCondition::BetweenUpperExclusive: {
       Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
       const auto value2 = boost::get<T>(*variant_value2);
-      return value >= max || value2 < min;
+      return value > max || value2 <= min;
     }
     case PredicateCondition::BetweenExclusive: {
       Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
```

Comment:
I think `.has_value()` is clearer than such a cast on an optional.

Reply:
Copied code. I'll just use operator bool.